python/lsst/cp/pipe/measureCrosstalk.py

```diff
@@ -0,0 +1,799 @@
+            DataIds to use to populate the output calibration.
+
+        Returns
+        -------
+        results : `lsst.pipe.base.Struct`
+            The results struct containing:
+
+            ``outputCrosstalk`` : `lsst.ip.isr.CrosstalkCalib`
+                Final crosstalk calibration.
+            ``outputProvenance`` : `lsst.ip.isr.IsrProvenance`
+                Provenance data for the new calibration.
+
+        Raises
+        ------
+        RuntimeError
+            Raised if the input data contains too many target
+            detectors.
+
+        Notes
+        -----
+        The lsstDebug.Info() method can be rewritten for __name__ =
+        `lsst.ip.isr.measureCrosstalk`, and supports the parameters:
+
+        debug.display['reduce'] : `bool`
+            Display a histogram of the combined ratio measurements for
+            a pair of source/target amplifiers from all input
+            exposures/detectors.
+
+        """
+        if outputDims:
+            calibChip = outputDims['detector']
+        else:
+            # calibChip needs to be set manually in Gen2.
+            calibChip = None
+
+        self.log.info("Combining measurements from %d ratios and %d fluxes",
+                      len(inputRatios), len(inputFluxes) if inputFluxes else 0)
+
+        if inputFluxes is None:
+            inputFluxes = [None for exp in inputRatios]
+
+        combinedRatios = defaultdict(lambda: defaultdict(list))
+        combinedFluxes = defaultdict(lambda: defaultdict(list))
+        for ratioDict, fluxDict in zip(inputRatios, inputFluxes):
+            for targetChip in ratioDict:
+                if calibChip and targetChip != calibChip:
+                    raise RuntimeError("Received too many target chips!")
+
+                sourceChip = targetChip
+                if sourceChip in ratioDict[targetChip]:
+                    ratios = ratioDict[targetChip][sourceChip]
+
+                    for targetAmp in ratios:
+                        for sourceAmp in ratios[targetAmp]:
+                            combinedRatios[targetAmp][sourceAmp].extend(ratios[targetAmp][sourceAmp])
+                            if fluxDict:
+                                combinedFluxes[targetAmp][sourceAmp].extend(fluxDict[sourceChip][sourceAmp])
+                # Iterating over all other entries in ratioDict[targetChip] will yield
+                # inter-chip terms.  Skipping to reduce complexity.
+
+        for targetAmp in combinedRatios:
+            for sourceAmp in combinedRatios[targetAmp]:
+                self.log.info("Read %d pixels for %s -> %s",
+                              len(combinedRatios[targetAmp][sourceAmp]),
+                              targetAmp, sourceAmp)
+                if len(combinedRatios[targetAmp][sourceAmp]) > 100:
+                    self.debugRatios('reduce', combinedRatios, targetAmp, sourceAmp, 0.0)
+
+        if self.config.fluxOrder == 0:
```

Comment:
If we have nonlinear crosstalk, we'll need to fit something, and a polynomial is the likely first choice.  The current method just generates the zero-order polynomial.  Since no one has thought much about nonlinear crosstalk implementations, this is mostly a placeholder.

Reply:
OK, fair enough.